File: file/dami.js
//toastlog("damizhibo")
const HOST = 'https://www.damizhibo.com';
const LIST_URL = 'https://www.damizhibo.com/list.json';
const LOGO = 'https://www.damizhibo.com/favicon.ico';
const UA = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/140.0.0.0 Safari/537.36';

/* ============================================================================
 *   工具函数
 * ============================================================================ */
function clean(s) {
  return String(s || '').replace(/<script[\s\S]*?<\/script>/gi, '')
    .replace(/<style[\s\S]*?<\/style>/gi, '')
    .replace(/<[^>]+>/g, '')
    .replace(/&nbsp;/g, ' ').replace(/&amp;/g, '&')
    .replace(/&quot;/g, '"').replace(/&#39;/g, "'")
    .replace(/\s+/g, ' ').trim();
}

function absUrl(url) {
  if (!url) return LOGO;
  url = String(url).trim();
  if (/^https?:\/\//i.test(url)) return url;
  if (url.indexOf('//') === 0) return 'https:' + url;
  if (url[0] === '/') return HOST.replace(/\/$/, '') + url;
  return HOST.replace(/\/$/, '') + '/' + url;
}

function safeJson(t, d) {
  try { return JSON.parse(t || '{}'); } catch(e) { return d || {}; }
}

function getHeaders() {
  return {
    'User-Agent': UA,
    'Referer': HOST + '/',
    'Accept': '*/*',
    'Accept-Language': 'zh-CN,zh;q=0.9'
  };
}

/* ── 网络请求 ── */
async function fetchText(url, timeout) {
  const hd = getHeaders();
  const to = timeout || 8000;
  try {
    if (typeof Java !== 'undefined' && Java && Java.req) {
      const r = await Java.req(url, { headers: hd, timeout: to });
      if (typeof r === 'string') return r;
      return String((r && (r.body || r.content || r.data)) || '');
    }
    const r2 = await req(url, { headers: hd, timeout: to });
    if (typeof r2 === 'string') return r2;
    return String((r2 && (r2.content || r2.body || r2.data)) || '');
  } catch(e) {
    return '';
  }
}

async function fetchJson(url) {
  try { return JSON.parse(await fetchText(url) || '{}'); } catch(e) { return {}; }
}

/* ── 球队Logo缓存 ── */
let _teamLogos = {};
let _logoLoaded = false;

/**
 * 从首页HTML解析球队Logo
 */
async function loadTeamLogos() {
  if (_logoLoaded) return;
  try {
    const html = await fetchText(HOST + '/?_t=' + Date.now());
    if (!html) return;

    // 解析 img 标签：div.team-logo.img-logo > img[src][alt]
    const imgRegex = /<img[^>]*src="([^"]*)"[^>]*alt="([^"]*)"[^>]*>/g;
    let m;
    while ((m = imgRegex.exec(html)) !== null) {
      const alt = m[2].trim();
      if (alt && alt.length > 1 && !alt.includes('logo') && !alt.includes('Logo')) {
        _teamLogos[alt] = m[1];
      }
    }

    _logoLoaded = true;
  } catch(e) {
    // 静默
  }
}

function getTeamLogo(teamName) {
  return _teamLogos[teamName] || '';
}

/* ── 比赛数据 ── */
async function fetchAllMatches() {
  try {
    const text = await fetchText(LIST_URL);
    if (!text) return [];
    const data = JSON.parse(text);
    if (Array.isArray(data)) return data;
    if (data.data && Array.isArray(data.data)) return data.data;
    if (data.list && Array.isArray(data.list)) return data.list;
    if (data.result && Array.isArray(data.result)) return data.result;
    return [];
  } catch(e) {
    return [];
  }
}

/* ── 分类判断 ── */
function isFootball(league, sportType) {
  if (sportType === 'football') return true;
  const name = (league || '').toLowerCase();
  const kw = [
    '足球','football','soccer',
    '世界杯','world cup','中超','chinese super','csl',
    '英超','premier','uefa','欧冠','欧联','europa',
    '西甲','laliga','德甲','bundesliga','意甲','serie a',
    '法甲','ligue 1','中甲','中乙','足协杯',
    '亚冠','afc','世预赛','欧国联','nations league',
    '日职','j league','韩k','k league','澳超','a league',
    '荷甲','eredivisie','葡超','primeira liga',
    '俄超','冰岛超','icelandic','爱超',
    '罗甲','捷甲','czech','匈甲','hungarian',
    '中冠','村超','哈萨甲','拉脱超','摩洛超',
    '埃塞超','白俄超','蒙古超','越女联','韩女联',
    '球会友谊','澳威超','澳维超','澳达超','巴西乙',
    '土超','比甲','丹超','挪超','瑞典超','奥超','瑞士超'
  ];
  for (const k of kw) { if (name.includes(k)) return true; }
  return false;
}

function isBasketball(league, sportType) {
  if (sportType === 'basketball') return true;
  const name = (league || '').toLowerCase();
  const kw = [
    '篮球','basketball',
    'nba','wnba','cba','nbl',
    '欧冠篮','euroleague','欧篮',
    '发展联盟','g league','ncaa',
    '新西兰联','nznbl','澳洲篮','nbl australia',
    '印尼联','ibl','日本篮','b league',
    '韩国篮','kbl','菲律宾篮','pba',
    '国际赛女','women international',
    '土篮超','bsl','西篮甲','德篮甲',
    '法篮甲','意篮甲','立陶宛联','lkl',
    'vbl','越南联','泰国联','马来西亚联'
  ];
  for (const k of kw) { if (name.includes(k)) return true; }
  return false;
}

function isHot(league) {
  const name = (league || '').toLowerCase();
  const kw = ['世界杯','world cup','中超','csl','wnba','nba','cba',
    '中冠','冰岛超','icelandic','巴西乙','国际赛'];
  for (const k of kw) { if (name.includes(k)) return true; }
  return false;
}

/* ── 构建VOD ── */
function buildVod(match) {
  if (!match) return null;
  const namiId = String(match.nami_id || match.id || match.match_id || '');
  const homeTeam = match.home_team || match.home || match.team1 || match.team_a || '';
  const awayTeam = match.away_team || match.away || match.team2 || match.team_b || '';
  const leagueName = match.league_name || match.league || match.tournament || match.series || '';
  const timeStr = match.match_time || match.time || match.start_time || match.date || '';

  // 格式化时间
  let remark = leagueName;
  if (timeStr) {
    const t = timeStr.replace(/T/, ' ').substring(0, 16);
    if (t.length >= 10) remark = leagueName + ' ' + t;
  }

  // 🟢 判断是否已开赛（比赛时间 ≤ 当前时间）
  if (timeStr) {
    const cleanTime = timeStr.replace(/T/, ' ').substring(0, 19);
    const matchTime = new Date(cleanTime);
    const now = new Date();
    if (!isNaN(matchTime.getTime()) && matchTime <= now) {
      remark = '🟢' + remark;
    }
  }

  // 主队Logo
  const logo = getTeamLogo(homeTeam);

  return {
    vod_id: 'dmz|' + namiId,
    vod_name: homeTeam + ' vs ' + awayTeam,
    vod_pic: logo || LOGO,
    vod_remarks: remark,
    vod_content: (leagueName || '体育直播') + '\n' + homeTeam + ' vs ' + awayTeam,
    type_name: leagueName || '全部'
  };
}

/* ── 分类定义 ── */
function getClasses() {
  return [
    { type_id: 'all', type_name: '全部' },
    { type_id: 'hot', type_name: '🔥 热门' },
    { type_id: 'football', type_name: '⚽ 足球' },
    { type_id: 'basketball', type_name: '🏀 篮球' }
  ];
}

/* ============================================================================
 *   爬虫接口
 * ============================================================================ */
async function init(cfg) { }

async function home(filter) {
  return JSON.stringify({ class: getClasses(), filters: {} });
}

async function homeVod() {
  return await category('all', 1, {}, {});
}

async function category(tid, pg, filter, extend) {
  tid = String((extend && extend.cateId) || tid || 'all');
  pg = parseInt(pg) || 1;
  let list = [];

  try {
    await loadTeamLogos();
    const matches = await fetchAllMatches();

    for (let i = 0; i < matches.length; i++) {
      const m = matches[i];
      if (!m) continue;
      const league = m.league_name || m.league || m.tournament || m.series || '';

      // 分类过滤
      if (tid === 'hot' && !isHot(league)) continue;
      if (tid === 'football' && !isFootball(league)) continue;
      if (tid === 'basketball' && !isBasketball(league)) continue;

      const vod = buildVod(m);
      if (vod) list.push(vod);
    }
  } catch(e) {}

  return JSON.stringify({
    code: 1, msg: '赛程列表', page: pg, pagecount: 1,
    limit: 200, total: list.length, list
  });
}

async function detail(ids) {
  const id = String(Array.isArray(ids) ? ids[0] : ids || '');
  const parts = id.split('|');
  const namiId = parts.length > 1 ? parts[1] : id;

  try {
    await loadTeamLogos();
    const matches = await fetchAllMatches();
    let target = null;

    for (let i = 0; i < matches.length; i++) {
      const m = matches[i];
      if (!m) continue;
      if (String(m.nami_id || m.id || m.match_id || '') === namiId) {
        target = m;
        break;
      }
    }

    if (!target) {
      return JSON.stringify({ code: 1, msg: 'ok', page: 1, pagecount: 1, limit: 1, total: 0, list: [] });
    }

    const homeTeam = target.home_team || target.home || target.team1 || target.team_a || '';
    const awayTeam = target.away_team || target.away || target.team2 || target.team_b || '';
    const leagueName = target.league_name || target.league || target.tournament || target.series || '';
    const signals = target.signals || target.live_signals || [];

    // 构建播放列表
    let playUrls = [];
    if (signals.length > 0) {
      for (let si = 0; si < signals.length; si++) {
        const sig = signals[si];
        const name = sig.label || sig.name || sig.title || '线路' + (si + 1);
        const url = sig.url || sig.link || sig.src || '';
        if (url) {
          playUrls.push(name + '$' + url);
        }
      }
    }

    if (playUrls.length === 0) {
      playUrls.push('暂无信号$' + HOST);
    }

    const vod = {
      vod_id: id,
      vod_name: homeTeam + ' vs ' + awayTeam,
      vod_pic: getTeamLogo(homeTeam) || LOGO,
      vod_remarks: leagueName,
      vod_content: (leagueName || '体育直播') + '\n' + homeTeam + ' vs ' + awayTeam,
      vod_play_from: '大米直播',
      vod_play_url: playUrls.join('#')
    };

    return JSON.stringify({ code: 1, msg: '直播详情', page: 1, pagecount: 1, limit: 1, total: 1, list: [vod] });
  } catch(e) {
    return JSON.stringify({ code: 1, msg: 'ok', page: 1, pagecount: 1, limit: 1, total: 0, list: [] });
  }
}

async function search(wd, quick, pg) {
  const keyword = (wd || '').trim().toLowerCase();
  if (!keyword) {
    return JSON.stringify({ code: 1, msg: '搜索', page: 1, pagecount: 1, limit: 20, total: 0, list: [] });
  }

  try {
    await loadTeamLogos();
    const matches = await fetchAllMatches();
    const list = [];

    for (let i = 0; i < matches.length; i++) {
      const m = matches[i];
      if (!m) continue;
      const homeTeam = m.home_team || m.home || m.team1 || m.team_a || '';
      const awayTeam = m.away_team || m.away || m.team2 || m.team_b || '';
      const league = m.league_name || m.league || m.tournament || m.series || '';
      const searchText = (homeTeam + ' ' + awayTeam + ' ' + league).toLowerCase();

      if (searchText.indexOf(keyword) === -1) continue;

      const vod = buildVod(m);
      if (vod) list.push(vod);
    }

    return JSON.stringify({ code: 1, msg: '搜索结果', page: pg || 1, pagecount: 1, limit: 50, total: list.length, list });
  } catch(e) {
    return JSON.stringify({ code: 1, msg: '搜索', page: 1, pagecount: 1, limit: 20, total: 0, list: [] });
  }
}

async function play(flag, id, flags) {
  const url = String(id || '');
  if (/\.(m3u8|flv|mp4)(\?|$)/i.test(url)) {
    return JSON.stringify({
      parse: 0, url: url,
      header: {
        'User-Agent': UA,
        'Referer': HOST + '/',
        'Origin': HOST
      }
    });
  }
  return JSON.stringify({
    parse: 1, url: url,
    header: {
      'User-Agent': UA,
      'Referer': HOST + '/',
      'Origin': HOST
    }
  });
}

/* ============================================================================
 *   包装函数（统一响应格式）
 * ============================================================================ */
async function homeContent(f) {
  return safeJson(await home(f), { class: [], filters: {} });
}

async function homeVideoContent() {
  return safeJson(await homeVod(), { list: [] });
}

async function categoryContent(tid, pg, f, ext) {
  return safeJson(await category(tid, pg, f, ext || {}), { list: [] });
}

async function detailContent(ids) {
  return safeJson(await detail(ids), { list: [] });
}

async function searchContent(wd, q, pg) {
  return safeJson(await search(wd, q, pg || 1), { list: [] });
}

async function playerContent(flag, id, flags) {
  return safeJson(await play(flag, id, flags), { parse: 1, url: id });
}

/* ============================================================================
 *   引擎导出 — 全局函数模式
 * ============================================================================ */
export function __jsEvalReturn() {
  return {
    init: init,
    home: home,
    homeVod: homeVod,
    category: category,
    search: search,
    detail: detail,
    play: play,
    homeContent: homeContent,
    homeVideoContent: homeVideoContent,
    categoryContent: categoryContent,
    detailContent: detailContent,
    searchContent: searchContent,
    playerContent: playerContent
  };
}

File: file/dami.py
#!/usr/bin/env python3
# 大米直播 V2.5 - 修复: 信号路数(live-scores.json实时比分+detail API信号补全)
import re, json, time, requests
from datetime import datetime, timezone, timedelta
from base.spider import Spider as BaseSpider

HOST = 'https://www.damizhibo.com'
LIST_URL = 'https://www.damizhibo.com/list.json'
LOGO = 'https://www.damizhibo.com/favicon.ico'
UA = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/140.0.0.0 Safari/537.36'

FOOTBALL_KW = ['足球','football','soccer','世界杯','world cup','中超','csl',
    '英超','premier','uefa','欧冠','欧联','europa','西甲','laliga','德甲',
    'bundesliga','意甲','serie a','法甲','ligue 1','中甲','中乙','足协杯',
    '亚冠','afc','世预赛','欧国联','日职','j league','韩k','k league',
    '澳超','a league','荷甲','葡超','俄超','冰岛超','爱超','罗甲','捷甲',
    '匈甲','中冠','哈萨甲','巴西乙','土超','比甲','丹超','挪超','瑞典超',
    '奥超','瑞士超','美甲','美乙','美女职','美青杯','联盟杯','墨联','墨西乙',
    '委内超','巴西杯','阿甲','秘鲁甲','玻利甲','菲州长杯','欧联杯','欧女杯',
    '韩女联','澳南超','澳新南联','澳维女超','韩K2联','球会友谊','白俄超',
    '东南锦','日职联','国际赛女','英联杯']

BASKETBALL_KW = ['篮球','basketball','nba','wnba','cba','nbl','欧冠篮',
    'euroleague','欧篮','g league','ncaa','pba','澳nbl','韩k2联']

HOT_KW = ['世界杯','world cup','中超','csl','wnba','nba','cba','中冠',
    '冰岛超','巴西乙','国际赛','联盟杯','欧联杯','欧足联','英联杯',
    'wnba','墨联','nbl','日职联','东南锦']


class Spider(BaseSpider):
    def getName(self):
        return "大米直播"

    def init(self, extend=""):
        self.host = HOST
        self._cache = {}
        self._CACHE_TTL = 300

    def _get_cache(self, key):
        e = self._cache.get(key)
        if e and time.time() - e['t'] < self._CACHE_TTL:
            return e['v']
        return None

    def _set_cache(self, key, val):
        self._cache[key] = {'v': val, 't': time.time()}

    def _headers(self):
        return {'User-Agent': UA, 'Referer': self.host + '/', 'Accept': '*/*'}

    def _html_headers(self):
        return {'User-Agent': UA, 'Referer': self.host + '/', 'Accept': 'text/html,application/xhtml+xml'}

    def _get_json(self, url, referer=None, timeout=15):
        try:
            hd = self._headers()
            if referer: hd['Referer'] = referer
            r = requests.get(url, headers=hd, timeout=timeout)
            return json.loads(r.text or '{}')
        except:
            return {}

    def _fmt_time(self, t):
        if not t: return ''
        try:
            t_clean = t[:16].strip()
            if len(t_clean) >= 16:
                return f'{t_clean[5:7]}-{t_clean[8:10]} {t_clean[11:16]}'
            return t_clean
        except:
            return ''

    def _get_sport_type(self, m):
        mt = m.get('match_type', '').lower()
        if mt: return mt
        league = m.get('league', '').lower()
        if self._match_type(league, BASKETBALL_KW): return 'basketball'
        if self._match_type(league, FOOTBALL_KW): return 'football'
        return 'other'

    def _match_type(self, league, kw_list):
        name = league.lower()
        return any(k in name for k in kw_list)

    def _parse_html(self, html):
        """从HTML解析: Logo映射 + 比赛卡片(比分+状态)"""
        if not html: return {}, []
        logos = {}
        for src, alt in re.findall(r'<img[^>]*src="([^"]+)"[^>]*alt="([^"]+)"[^>]*>', html):
            alt = alt.strip()
            if alt and len(alt) > 1 and 'logo' not in alt.lower():
                logos[alt] = src if src.startswith('http') else HOST + '/' + src.lstrip('/')
        cards = []
        for m in re.finditer(r'<a[^>]*class="match-card[^"]*"[^>]*>(.*?)</a>', html, re.DOTALL):
            tag_text = m.group(0)[:m.group(0).find('>') + 1]
            card = m.group(1)
            mid_m = re.search(r'data-match-id="([^"]*)"', tag_text)
            match_id = mid_m.group(1).strip() if mid_m else ''
            mt_m = re.search(r'data-match-type="([^"]*)"', tag_text)
            match_type = mt_m.group(1).strip() if mt_m else ''
            dt_m = re.search(r'data-match-time="([^"]*)"', tag_text)
            data_time = dt_m.group(1).strip() if dt_m else ''
            league_m = re.search(r'class="league-badge"[^>]*>([^<]+)', card)
            league = league_m.group(1).strip() if league_m else ''
            time_m = re.search(r'class="match-time"[^>]*>([^<]+)', card)
            time_str = time_m.group(1).strip() if time_m else ''
            status_m = re.search(r'class="match-status[^"]*">([^<]+)', card)
            status = status_m.group(1).strip() if status_m else ''
            home_m = re.search(r'team-home.*?class="team-name">([^<]+)', card, re.DOTALL)
            home = home_m.group(1).strip() if home_m else ''
            away_m = re.search(r'team-away.*?class="team-name">([^<]+)', card, re.DOTALL)
            away = away_m.group(1).strip() if away_m else ''
            score_m = re.search(r'class="score">([^<]+)', card)
            score = score_m.group(1).strip().replace(' ', '') if score_m else ''
            home_logo_m = re.search(r'team-home.*?src="([^"]+)"', card, re.DOTALL)
            away_logo_m = re.search(r'team-away.*?src="([^"]+)"', card, re.DOTALL)
            home_logo = home_logo_m.group(1) if home_logo_m else ''
            away_logo = away_logo_m.group(1) if away_logo_m else ''
            cards.append({
                'match_id': match_id, 'league': league, 'time_str': time_str, 'data_time': data_time,
                'match_type': match_type, 'status': status, 'home': home, 'away': away, 'score': score,
                'home_logo': home_logo, 'away_logo': away_logo,
            })
        return logos, cards

    def _load_all_data(self):
        cached = self._get_cache('all_data')
        if cached is not None: return cached

        list_data = self._get_json(LIST_URL)
        matches_list = list_data if isinstance(list_data, list) else (list_data.get('data') or list_data.get('list') or [])

        # nami_id -> info (精确映射)
        id_map = {}
        for m in matches_list:
            nid = str(m.get('nami_id') or m.get('id') or '')
            if nid:
                id_map[nid] = {
                    'nami_id': nid, 'home': m.get('home_team', ''), 'away': m.get('away_team', ''),
                    'league': m.get('league_name', ''), 'time': m.get('match_time', ''),
                    'signals': m.get('signals', []),
                }

        live_scores = {}
        try:
            ls_data = self._get_json(f'{HOST}/api/live-scores.json?_={int(time.time()*1000)}')
            live_scores = ls_data.get('scores', {})
        except:
            pass

        try:
            r = requests.get(HOST + '/', headers=self._html_headers(), timeout=15)
            r.encoding = r.apparent_encoding or 'utf-8'
            html = r.text
        except:
            html = ''

        logos, html_cards = self._parse_html(html)

        result = []
        matched_ids = set()
        for card in html_cards:
            mid = card['match_id']
            info = id_map.get(mid, {})
            if mid and mid not in matched_ids:
                matched_ids.add(mid)
            elif not mid:
                for nid, v in id_map.items():
                    if v['home'] == card['home'] and v['away'] == card['away'] and nid not in matched_ids:
                        info = v
                        mid = nid
                        matched_ids.add(nid)
                        break
            home_logo = card['home_logo'] or logos.get(card['home'], '') or info.get('home_logo', '')
            away_logo = card['away_logo'] or logos.get(card['away'], '') or info.get('away_logo', '')
            time_val = card['data_time'] or info.get('time', '')
            nid = info.get('nami_id', mid)
            score = card['score']
            status = card['status']
            if nid and nid in live_scores:
                ls = live_scores[nid]
                hs = ls.get('home_score', '')
                aws = ls.get('away_score', '')
                if hs != '' and aws != '': score = f'{hs}-{aws}'
                ls_status = ls.get('status', '')
                if ls_status == 'live': status = '直播中'
                elif ls_status == 'finished': status = '完场'
            result.append({
                'nami_id': nid, 'home': card['home'], 'away': card['away'],
                'league': card['league'] or info.get('league', ''),
                'time': time_val, 'match_type': card.get('match_type', ''),
                'status': status, 'score': score,
                'home_logo': home_logo, 'away_logo': away_logo,
                'signals': info.get('signals', []),
            })

        for nid, info in id_map.items():
            if nid not in matched_ids:
                score = ''
                status = 'unknown'
                if nid in live_scores:
                    ls = live_scores[nid]
                    hs = ls.get('home_score', '')
                    aws = ls.get('away_score', '')
                    if hs != '' and aws != '': score = f'{hs}-{aws}'
                    ls_status = ls.get('status', '')
                    if ls_status == 'live': status = '直播中'
                    elif ls_status == 'finished': status = '完场'
                result.append({
                    'nami_id': nid, 'home': info['home'], 'away': info['away'],
                    'league': info['league'], 'time': info['time'],
                    'match_type': '', 'status': status, 'score': score,
                    'home_logo': logos.get(info['home'], ''),
                    'away_logo': logos.get(info['away'], ''),
                    'signals': info['signals'],
                })

        self._set_cache('all_data', result)
        return result

    def _fetch_detail_signals(self, nid):
        try:
            data = self._get_json(f'{HOST}/api/live/{nid}.json', referer=f'{HOST}/live/{nid}.html')
            sigs = data.get('signals', [])
            if sigs:
                labels = set()
                for s in sigs:
                    label = s.get('label', '')
                    if label in ('高清K', '高清J', '高清B'):
                        labels.add(label)
                return sigs
        except:
            pass
        return None

    def _build_name(self, home, away, score):
        if score and score != '-':
            return f'{home} {score} {away}'
        return f'{home} vs {away}'

    def _build_remarks(self, m):
        is_live = m.get('status') == '直播中'
        tag = '\U0001f534' if is_live else ''
        dt = self._fmt_time(m.get('time', ''))
        league = m.get('league', '')
        sig_count = len(m.get('signals', []))
        parts = []
        if tag: parts.append(tag)
        if dt: parts.append(dt)
        if league: parts.append(league)
        if sig_count >= 2: parts.append(f'{sig_count}路')
        return ' '.join(parts) or '直播'

    def _get_classes(self):
        return [
            {'type_id': 'all', 'type_name': '全部'},
            {'type_id': 'hot', 'type_name': '🔥 热门'},
            {'type_id': 'football', 'type_name': '⚽ 足球'},
            {'type_id': 'basketball', 'type_name': '🏀 篮球'},
        ]

    def homeContent(self, filter):
        return {'class': self._get_classes(), 'filters': {}}

    def categoryContent(self, tid, pg, filter, extend):
        try:
            tid = (extend or {}).get('cateId') or tid or 'all'
            pg = int(pg or 1)
            all_data = self._load_all_data()
            list_items = []
            for m in all_data:
                league = m.get('league', '')
                sport = self._get_sport_type(m)
                if tid == 'hot' and not self._match_type(league, HOT_KW): continue
                if tid == 'football' and sport != 'football': continue
                if tid == 'basketball' and sport != 'basketball': continue
                nid = m.get('nami_id', '')
                vod_id = f'dmz|{nid}' if nid else f'dmz|{m["home"]}_{m["away"]}'
                pic = m.get('home_logo', '') or m.get('away_logo', '') or LOGO
                list_items.append({
                    'vod_id': vod_id,
                    'vod_name': self._build_name(m['home'], m['away'], m.get('score', '')),
                    'vod_pic': pic,
                    'vod_remarks': self._build_remarks(m),
                    'type_name': league or '全部',
                })
            return {
                'page': pg, 'pagecount': 1, 'limit': len(list_items),
                'total': len(list_items), 'list': list_items
            }
        except:
            return {'page': 1, 'pagecount': 1, 'limit': 30, 'total': 0, 'list': []}

    def detailContent(self, ids):
        try:
            vid = (ids[0] if isinstance(ids, list) else str(ids or '')).strip()
            if not vid: return {'list': []}
            parts = vid.split('|')
            nid = parts[1] if len(parts) > 1 else parts[0]
            all_data = self._load_all_data()
            target = None
            for m in all_data:
                if m.get('nami_id') == nid or f'{m["home"]}_{m["away"]}' == nid:
                    target = m
                    break
            if not target: return {'list': []}
            home = target['home']
            away = target['away']
            league = target.get('league', '')
            signals = target.get('signals', [])
            if nid and len(signals) < 3:
                detail_sigs = self._fetch_detail_signals(nid)
                if detail_sigs and len(detail_sigs) > len(signals):
                    signals = detail_sigs
            play_urls = []
            for sig in signals:
                label = sig.get('label') or sig.get('name') or sig.get('title') or ''
                url = sig.get('url') or sig.get('link') or sig.get('src') or ''
                if url: play_urls.append(f'{label}${url}')
            if not play_urls:
                play_urls.append(f'暂无信号${HOST}')
            pic = target.get('home_logo', '') or target.get('away_logo', '') or LOGO
            return {'list': [{
                'vod_id': vid,
                'vod_name': self._build_name(home, away, target.get('score', '')),
                'vod_pic': pic,
                'vod_remarks': self._build_remarks({**target, 'signals': signals}),
                'vod_play_from': '大米直播',
                'vod_play_url': '#'.join(play_urls),
                'vod_content': (league or '体育直播') + '\n' + self._build_name(home, away, ''),
            }]}
        except:
            return {'list': []}

    def searchContent(self, kw, quick, pg):
        try:
            keyword = (kw or '').strip().lower()
            if not keyword:
                return {'page': 1, 'pagecount': 1, 'limit': 20, 'total': 0, 'list': []}
            all_data = self._load_all_data()
            list_items = []
            for m in all_data:
                if keyword in (m['home'] + ' ' + m['away'] + ' ' + m.get('league', '')).lower():
                    nid = m.get('nami_id', '')
                    pic = m.get('home_logo', '') or m.get('away_logo', '') or LOGO
                    list_items.append({
                        'vod_id': f'dmz|{nid}' if nid else f'dmz|{m["home"]}_{m["away"]}',
                        'vod_name': self._build_name(m['home'], m['away'], m.get('score', '')),
                        'vod_pic': pic,
                        'vod_remarks': self._build_remarks(m),
                    })
            return {
                'page': pg or 1, 'pagecount': 1, 'limit': len(list_items),
                'total': len(list_items), 'list': list_items
            }
        except:
            return {'page': 1, 'pagecount': 1, 'limit': 20, 'total': 0, 'list': []}

    def playerContent(self, flag, id, vipFlags):
        try:
            url = str(id or '')
            return {'parse': 0, 'url': url, 'header': {
                'User-Agent': UA, 'Referer': HOST + '/', 'Origin': HOST
            }}
        except:
            return {'parse': 0, 'url': str(id or '')}

    def localProxy(self, param):
        return [200, "text/plain", b"", {}]